### PR TITLE
Axis Bid Adapter : initial release

### PR DESCRIPTION
**Pull request by PyjamaWarrior**
doc - https://github.com/prebid/prebid.github.io/pull/4424

### CODE REVIEW

**Comment by Rothalack** [comment]
Hey @PyjamaWarrior I think your code here is good. I'm just concerned with your endpoint. When I test locally I am getting the error Prebid axis ERROR: Server call for axis failed:  0. Continuing without bids. Are you able to test with the integration tests on your end?

I believe this is because my local test is on http, not https, not certain though. I am also seeing 1000+ms response time from your server as well. Is this intended?

**Comment by Rothalack** [comment]
Hey @PyjamaWarrior , just wanted to check in and see if you had any questions or anything. Are you still working on things on your end?

File: modules/axisBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      deviceHeight,
+      language,
+      secure,
+      host,
+      page,
+      placements,
+      coppa: config.getConfig('coppa') === true ? 1 : 0,
+      ccpa: bidderRequest.uspConsent || undefined,
+      gdpr: bidderRequest.gdprConsent || undefined,
+      tmax: config.getConfig('bidderTimeout')
```
Comment (patmmccann) [inline modules/axisBidAdapter.js:152]: please prefer the request object to config

File: modules/axisBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    const placements = [];
+    const request = {
+      deviceWidth,
+      deviceHeight,
+      language,
+      secure,
+      host,
+      page,
+      placements,
+      coppa: config.getConfig('coppa') === true ? 1 : 0,
```
Comment (patmmccann) [inline modules/axisBidAdapter.js:149]: please prefer the request object to config

**Comment by PyjamaWarrior** [comment]
@Rothalack  Can you check again, please? Now everything seems to be working fine.

**Comment by Rothalack** [comment]
I'll have a look today. In the meantime, please try to address Pat's feedback above. I should have time to test things in an hour or two.

**Comment by Rothalack** [comment]
@PyjamaWarrior I got a chance to test, everything looks good and is working. I still think you should address Pat's feedback about using the request instead of config above.

**Comment by patmmccann** [comment]
Thanks so much @Rothalack ! I'll leave assigned to me to get this last change over the finish line.

File: modules/axisBidAdapter.js (hunk 2 above)
Comment (patmmccann) [reply modules/axisBidAdapter.js:153]: @PyjamaWarrior here is an example 

https://github.com/prebid/Prebid.js/blob/509deaa0649824369907e6b76d0a4d045ec84df8/modules/riseBidAdapter.js#L328

File: modules/axisBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/axisBidAdapter.js:156]: @PyjamaWarrior here is an example https://github.com/prebid/Prebid.js/blob/509deaa0649824369907e6b76d0a4d045ec84df8/modules/riseBidAdapter.js#L421

**Review by patmmccann — CHANGES_REQUESTED** [review]
Please change these two locations to look at the request object instead of the configuration.

**Review by patmmccann — CHANGES_REQUESTED** [review]
We don't allow parameters that are just standard request fields

**Review by patmmccann — CHANGES_REQUESTED** [review]
Two changes are currently required: 

1) iabCat: ['IAB1-1', 'IAB3-1', 'IAB4-3'] is not an allowed parameter, please read this from ortb2.site.cat or pagecat. Publishers don't enjoy setting the same field in specific ways for each adapter. Feel free to include instructions on how publishers can set this parameter or link to https://docs.prebid.org/features/firstPartyData.html

2) You are failing your own test according to the last circleci run

**Comment by PyjamaWarrior** [comment]
@patmmccann I've fixed tests and iabCat. But i want to ask you about ortb2.site.cat. As you can see, i got this data from bidderRequest as you suggested and its same for all adUnits. Maybe its better to use ortb2Imp from validBidRequests to have categories for every adUnit separately? What do you think about that?

**Comment by patmmccann** [comment]
IAB hangs it on the site object, but we allow it to change per request. see [ Supplying Auction-Specific Data](https://docs.prebid.org/features/firstPartyData.html#supplying-adunit-specific-data)

**Review by patmmccann — APPROVED** [review]

**Pull request merged**